fr/lessons/basics/functions.md

```diff
@@ -0,0 +1,223 @@
+---
+layout: page
+title: Fonctions
+category: basics
+order: 7
+lang: fr
+---
+
+En Elixir et dans d'autres langages fonctionnels, les fonctions sont des citoyens de premier ordre. Nous verrons les différents types de fonctions en
+Elixir, ce qui les rend différentes et comment les utiliser.
+
+## Table des matières
+
+- [Fonctions anonymes](#fonctions-anonymes)
+  - [Le raccourcis &](#le-raccourcis--)
+- [Pattern matching](#pattern-matching)
+- [Fonctions nommées](#fonctions-nommees)
+  - [Fonctions Privées](#fonctions-privee)
+  - [Guards](#guards)
+  - [Arguments par défaut](#arguments-par-defauts)
+
+## Fonctions anonymes
+
+Tout comme leur nom le sous-entend, les fonctions anonymes n'ont pas de nom. Tel que nous l'avons vu dans la leçon `Enum`, elles sont fréquemment passées à
+d'autres fonctions. Pour définir une fonctions anonyme en Elixir, nous avons besoin des mot-clés `fn` et `end`, à l'intérieur desquels nous pouvons définir 
+n'importe quel nombre de paramètres et de corps de fonction, séparés par `->`.
+Jetons un coup d'œil à cet exemple : 
+
+```elixir
+iex> sum = fn (a, b) -> a + b end
+iex> sum.(2, 3)
+5
+```
+### Le raccourci '&'
+
+Utiliser des fonctions anonymes est quelque chose de tellement courant en Elixir qu'il y a un raccourci pour le faire : 
+
+```elixir
+iex> sum = &(&1 + &2)
+iex> sum.(2, 3)
+5
+```
+
+Comme vous avez peut-être déjà deviné, dans la version raccourcie nos paramètres sont disponibles en tant que `&1`, `&2`, `&3`, etc.
+
+## Pattern matching
+
+Le Pattern matching (ou « Filtrage par motif » en Français) en Elixir ne se limite pas juste aux variables. Il peut être appliqué aux signatures de
+fonctions, comme nous allons le voir dans cette section : 
+
+Elixir utilise le pattern matching pour identifier le premier groupe de paramètres correspondant, et appelle le corps de fonction correspondant : 
```

Comment:
Si, le français spécifie les espacements à gauche et à droite des doubles ponctuations (; : ? !)

Le 27 mars 2016 16:15:37 GMT+02:00, Julien Blanchard <notifications@github.com> a écrit :
>> +Utiliser des fonctions anonymes est quelque chose de tellement
>courant en Elixir qu'il y a un raccourci pour le faire : 
>> +
>> +```elixir
>> +iex> sum = &(&1 + &2)
>> +iex> sum.(2, 3)
>> +5
>> +```
>> +
>> +Comme vous avez peut-être déjà deviné, dans la version raccourcie
>nos paramètres sont disponibles en tant que `&1`, `&2`, `&3`, etc.
>> +
>> +## Pattern matching
>> +
>> +Le Pattern matching (ou « Filtrage par motif » en Français) en
>Elixir ne se limite pas juste aux variables. Il peut être appliqué aux
>signatures de
>> +fonctions, comme nous allons le voir dans cette section : 
>> +
>> +Elixir utilise le pattern matching pour identifier le premier groupe
>de paramètres correspondant, et appelle le corps de fonction
>correspondant : 
>
>pas d'espace avant ':'
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/doomspork/elixir-school/pull/229/files/f3c1059a0f4aa62904b176d27cfb1cadc5167a86#r57529171

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.